subsys/bluetooth/host/cs.c

```diff
@@ -561,6 +856,13 @@ void bt_hci_le_cs_test_end_complete(struct net_buf *buf)
 		return;
 	}
 
+	struct net_buf *reassembly_buf = get_reassembly_buf(BT_HCI_LE_CS_TEST_CONN_HANDLE, false);
+
+	if (reassembly_buf) {
+		LOG_WRN("De-allocating a dangling reassembly buffer");
```

Comment:
Controller should send pending CS subevent results to host in this case before we get the CS test end event

Reply:
Discussed offline: Will leave this check here as it's detecting an invalid behavior. Will add something similar to CS Procedure Enable Complete event when it's added.